pkg/apis/duck/v1beta1/channelable_types.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/apis"
+	"knative.dev/pkg/apis/duck"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+// +genduck
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Channelable is a skeleton type wrapping Subscribable and Addressable in the manner we expect resource writers
+// defining compatible resources to embed it. We will typically use this type to deserialize
+// Channelable ObjectReferences and access their subscription and address data.  This is not a real resource.
+type Channelable struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the part where the Channelable fulfills the Subscribable contract.
+	Spec ChannelableSpec `json:"spec,omitempty"`
+
+	Status ChannelableStatus `json:"status,omitempty"`
+}
+
+// ChannelableSpec contains Spec of the Channelable object
+type ChannelableSpec struct {
+	Subscribable `json:",inline"`
+
+	// DeliverySpec contains options controlling the event delivery
+	// +optional
+	Delivery *DeliverySpec `json:"delivery,omitempty"`
```

Comment:
Should Delivery be inside of Subscribable?

Reply:
So, currently the SubscriberSpec has a DeadLetterSinkURI in there, which is apis.URL. I wonder if that should be replaced with a Delivery? Otherwise, I don't see how the retry policy can with the current semantics be wired through to the Subscribable? So, that seems to mean we should put it in SubscriberSpec, thoughts?
Basically SubscriberSpec today has:
	// +optional
	DeadLetterSinkURI *apis.URL `json:"deadLetterSinkUri,omitempty"`

Seems that should be replaced with:
     Delivery *DeliverySpec `json:"deliver, omitempty"`

@lionelvillard @n3wscott

Reply:
I like this idea.

Reply:
ok, I'll bang it out.